### ISSUE TITLE
How to infer ONNX models using C++?

### ISSUE THREAD

**Issue by liegnaioo7** [ultralytics/ultralytics#25059]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

C++如何推理onnx的模型呢？
情景：我用yolo26训练并导出了onnx格式的分类模型，使用python推理全部正确，但是c++推理，结果却很不理想，有相关文档说明嘛？谢谢大家
How to infer ONNX models using C++?
Scenario: I trained and exported an ONNX classification model using YOLOv26. Inference using Python is completely correct, but inference using C++ yields unsatisfactory results. Is there any relevant documentation explaining this? Thank you.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @liegnaioo7, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you started, and an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for YOLO26 users, where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report related to C++ ONNX inference results differing from Python, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it. This should include your export command, model/task details, preprocessing and postprocessing steps used in C++, and a small sample input/output comparison if possible.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can check this:

https://github.com/ultralytics/ultralytics/tree/main/examples/cpp/ONNXRuntime

**Comment by liegnaioo7** [comment]
> You can check this:
> 
> https://github.com/ultralytics/ultralytics/tree/main/examples/cpp/ONNXRuntime

thanks！！！

**Comment by liegnaioo7** [comment]
> You can check this:
> 
> https://github.com/ultralytics/ultralytics/tree/main/examples/cpp/ONNXRuntime

its  worded！thank u verymuch！

**Comment by glenn-jocher** [comment]
不客气，很高兴已经解决！如果后续 C++ 推理结果和 Python 仍有差异，优先检查预处理是否一致（RGB/BGR、resize、NCHW/BCHW、`float32`、`/255.0` 归一化）以及分类输出的 softmax/top-k 解析。
